src/wallet/wallet.cpp

```diff
@@ -639,9 +639,7 @@ bool CWallet::EncryptWallet(const SecureString& strWalletPassphrase)
         if (IsHDEnabled()) {
             CKey key;
             CPubKey masterPubKey = GenerateNewHDMasterKey();
-            // preserve the old chains version to not break backward compatibility
-            CHDChain oldChain = GetHDChain();
-            if (!SetHDMasterKey(masterPubKey, &oldChain))
+            if (!SetHDMasterKey(masterPubKey))
```

Comment:
suggestion only: combine this with the line above to:

`if (!SetHDMasterKey(GenerateNewHDMasterKey()))`

Also: curly braces for single line if block please.